ksso/aws.py

```diff
@@ -0,0 +1,127 @@
+import json
+import logging
+from sys import exit
+
+import boto3
+from botocore.exceptions import BotoCoreError, ClientError
+
+logger = logging.getLogger("error")
+
+
+def assume_aws_role_with_keycloak_token(
+    token, aws_role_arn, session_name, session_duration_hours=2
+):
+    """
+    Assume an AWS IAM role using a Keycloak-provided token.
+
+    Args:
+        token (str): The Keycloak access token (WebIdentityToken).
+        aws_role_arn (str): The Amazon Resource Name (ARN) of the role to
+                            assume.
+        session_name (str): A name to uniquely identify the assumed session.
+
+    Returns:
+        Optional[Dict[str, str]]: AWS temporary credentials including
```

Comment:
LEt's add `typing` usage and annotate incoming and outcoming values to all funcs?

Reply:
Done